tests/phpunit/integration/Modules/Reader_Revenue_ManagerTest.php

```diff
@@ -73,4 +98,76 @@ public function test_service_classes_exist() {
 			class_exists( 'Google\Site_Kit_Dependencies\Google\Service\SubscribewithGoogle' )
 		);
 	}
+
+	public function test_get_datapoints() {
+		$this->assertEqualSets(
+			array(
+				'publications',
+			),
+			$this->reader_revenue_manager->get_datapoints()
+		);
+	}
+
+	public function test_get_publications() {
+		FakeHttp::fake_google_http_handler(
+			$this->reader_revenue_manager->get_client(),
+			function ( Request $request ) {
+				$url = parse_url( $request->getUri() );
+
+				switch ( $url['path'] ) {
+					case '/v1/publications':
+						$publications = array(
+							array(
+								'publicationId'         => 'ABCDEFGH',
+								'publicationPredicates' => array(
+									'businessPredicates' => array(
+										'supportsSiteKit' => true,
+										'canSell'         => true,
+									),
+								),
+								'verifiedDomains'       => 'example.com',
+								'paymentOptions'        => array(
+									'subscriptions' => true,
+									'noPayment'     => false,
+									'contributions' => false,
+									'thankStickers' => true,
+								),
+								'displayName'           => 'Test Property',
+								'products'              => array(
+									array(
+										'name' => 'basic',
+									),
+								),
+								'onboardingState'       => 'PENDING_VERIFICATION',
+							),
+						);
+						$response     = new ListPublicationsResponse();
+						$response->setPublications( $publications );
+						return new Response( 200, array(), json_encode( $response ) );
+				}
+			}
+		);
+
+		$this->reader_revenue_manager->register();
+
+		$this->authentication->get_oauth_client()->set_granted_scopes(
+			$this->authentication->get_oauth_client()->get_required_scopes()
+		);
+
+		$result = $this->reader_revenue_manager->get_data( 'publications' );
+		$this->assertNotWPError( $result );
+		$this->assertContainsOnlyInstancesOf( Publication::class, $result );
+
+		$publication = $result[0];
+
+		$this->assertEquals( 'Test Property', $publication->getDisplayName() );
+		$this->assertEquals( 'ABCDEFGH', $publication->getPublicationId() );
+	}
+
+	/**
+	 * @return Reader_Revenue_Manager
```

Comment:
We usually reference the `Module_With_Service_Entity` class here.

```suggestion
	 * @return Module_With_Service_Entity
```